src/python/grpcio_tests/tests_aio/unit/interceptor_test.py

```diff
@@ -0,0 +1,556 @@
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+
+            await server.stop(None)
+
+            call = multicallable(messages_pb2.SimpleRequest(), timeout=0.1)
+
+            with self.assertRaises(aio.AioRpcError) as exception_context:
+                await call
+
+            self.assertTrue(call.done())
+            self.assertFalse(call.cancelled())
+            self.assertEqual(await call.code(),
+                             grpc.StatusCode.DEADLINE_EXCEEDED)
+            self.assertEqual(await call.details(), 'Deadline Exceeded')
+            self.assertEqual(await call.initial_metadata(), ())
+            self.assertEqual(await call.trailing_metadata(), ())
+
+    async def test_cancel_before_rpc(self):
+
+        interceptor_reached = asyncio.Event()
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+                interceptor_reached.set()
+                await asyncio.sleep(0)
+
+                # This line should never be reached
+                raise Exception()
+
+        server_target, _ = await start_test_server()  # pylint: disable=unused-variable
+
+        async with aio.insecure_channel(server_target,
+                                        interceptors=[Interceptor()
+                                                     ]) as channel:
+
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+            call = multicallable(messages_pb2.SimpleRequest())
+
+            self.assertFalse(call.cancelled())
+            self.assertFalse(call.done())
+
+            await interceptor_reached.wait()
+            self.assertTrue(call.cancel())
+
+            with self.assertRaises(asyncio.CancelledError):
+                await call
+
+            self.assertTrue(call.cancelled())
+            self.assertTrue(call.done())
+            self.assertEqual(await call.code(), grpc.StatusCode.CANCELLED)
+            self.assertEqual(await call.details(),
+                             _LOCAL_CANCEL_DETAILS_EXPECTATION)
+            self.assertEqual(await call.initial_metadata(), None)
+            self.assertEqual(await call.trailing_metadata(), None)
+
+    async def test_cancel_after_rpc(self):
+
+        interceptor_reached = asyncio.Event()
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+                call = await continuation(client_call_details, request)
+                await call
+                interceptor_reached.set()
+                await asyncio.sleep(0)
```

Comment:
Same as above. If cancellation failed to be injected in this line, even without the `raise Exception()` below, the test seems will fail as well due to final status being `OK`.